File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/service/impl/BookStoreImpl.java
package io.micronaut.jimmer.java.it.service.impl;

import io.micronaut.jimmer.java.it.entity.BookStore;
import io.micronaut.jimmer.java.it.entity.BookStoreTable;
import io.micronaut.jimmer.java.it.entity.Fetchers;
import io.micronaut.jimmer.java.it.entity.Tables;
import io.micronaut.jimmer.java.it.service.IBookStore;
import jakarta.inject.Singleton;
import java.util.List;
import org.babyfish.jimmer.sql.JSqlClient;

@Singleton
public class BookStoreImpl implements IBookStore {

    BookStoreTable table = Tables.BOOK_STORE_TABLE;

    private final JSqlClient sqlClient;

    public BookStoreImpl(JSqlClient sqlClient) {
        this.sqlClient = sqlClient;
    }

    @Override
    public List<BookStore> oneToMany() {
        return sqlClient
                .createQuery(table)
                .select(
                        table.fetch(
                                Fetchers.BOOK_STORE_FETCHER
                                        .allScalarFields()
                                        .books(Fetchers.BOOK_FETCHER.allScalarFields())))
                .execute();
    }
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/service/impl/BookStoreImpl.kt
package io.micronaut.jimmer.kotlin.it.service.impl

import io.micronaut.jimmer.kotlin.it.entity.BookStore
import io.micronaut.jimmer.kotlin.it.entity.by
import io.micronaut.jimmer.kotlin.it.service.IBookStore
import jakarta.inject.Singleton
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.fetcher.newFetcher

@Singleton
class BookStoreImpl(
    private val sqlClient: KSqlClient,
) : IBookStore {
    override fun oneToMany(): List<BookStore> =
        sqlClient
            .createQuery(BookStore::class) {
                select(
                    table.fetch(
                        newFetcher(BookStore::class).by {
                            allScalarFields()
                            books {
                                allScalarFields()
                            }
                        },
                    ),
                )
            }.execute()
}
